// @vercel/sdk/esm/models/deleteredirectsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const DeleteRedirectsRequestBody$inboundSchema = z.object({
    name: z.string().optional(),
    redirects: z.array(z.string()),
});
/** @internal */
export const DeleteRedirectsRequestBody$outboundSchema = z.object({
    name: z.string().optional(),
    redirects: z.array(z.string()),
});
export function deleteRedirectsRequestBodyToJSON(deleteRedirectsRequestBody) {
    return JSON.stringify(DeleteRedirectsRequestBody$outboundSchema.parse(deleteRedirectsRequestBody));
}
export function deleteRedirectsRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteRedirectsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteRedirectsRequestBody' from JSON`);
}
/** @internal */
export const DeleteRedirectsRequest$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => DeleteRedirectsRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const DeleteRedirectsRequest$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => DeleteRedirectsRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function deleteRedirectsRequestToJSON(deleteRedirectsRequest) {
    return JSON.stringify(DeleteRedirectsRequest$outboundSchema.parse(deleteRedirectsRequest));
}
export function deleteRedirectsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteRedirectsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteRedirectsRequest' from JSON`);
}
/** @internal */
export const DeleteRedirectsVersion$inboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
/** @internal */
export const DeleteRedirectsVersion$outboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
export function deleteRedirectsVersionToJSON(deleteRedirectsVersion) {
    return JSON.stringify(DeleteRedirectsVersion$outboundSchema.parse(deleteRedirectsVersion));
}
export function deleteRedirectsVersionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteRedirectsVersion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteRedirectsVersion' from JSON`);
}
/** @internal */
export const DeleteRedirectsResponseBody$inboundSchema = z.object({
    alias: z.nullable(z.string()),
    version: z.lazy(() => DeleteRedirectsVersion$inboundSchema),
});
/** @internal */
export const DeleteRedirectsResponseBody$outboundSchema = z.object({
    alias: z.nullable(z.string()),
    version: z.lazy(() => DeleteRedirectsVersion$outboundSchema),
});
export function deleteRedirectsResponseBodyToJSON(deleteRedirectsResponseBody) {
    return JSON.stringify(DeleteRedirectsResponseBody$outboundSchema.parse(deleteRedirectsResponseBody));
}
export function deleteRedirectsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteRedirectsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteRedirectsResponseBody' from JSON`);
}
