// @vercel/sdk/esm/models/editredirectop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const Redirect$inboundSchema = z.object({
    source: z.string(),
    destination: z.string().optional(),
    statusCode: z.number().optional(),
    permanent: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    query: z.boolean().optional(),
});
/** @internal */
export const Redirect$outboundSchema = z.object({
    source: z.string(),
    destination: z.string().optional(),
    statusCode: z.number().optional(),
    permanent: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    query: z.boolean().optional(),
});
export function redirectToJSON(redirect) {
    return JSON.stringify(Redirect$outboundSchema.parse(redirect));
}
export function redirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Redirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Redirect' from JSON`);
}
/** @internal */
export const EditRedirectRequestBody$inboundSchema = z.object({
    name: z.string().optional(),
    redirect: z.lazy(() => Redirect$inboundSchema),
    restore: z.boolean().optional(),
});
/** @internal */
export const EditRedirectRequestBody$outboundSchema = z.object({
    name: z.string().optional(),
    redirect: z.lazy(() => Redirect$outboundSchema),
    restore: z.boolean().optional(),
});
export function editRedirectRequestBodyToJSON(editRedirectRequestBody) {
    return JSON.stringify(EditRedirectRequestBody$outboundSchema.parse(editRedirectRequestBody));
}
export function editRedirectRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditRedirectRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditRedirectRequestBody' from JSON`);
}
/** @internal */
export const EditRedirectRequest$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => EditRedirectRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const EditRedirectRequest$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => EditRedirectRequestBody$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function editRedirectRequestToJSON(editRedirectRequest) {
    return JSON.stringify(EditRedirectRequest$outboundSchema.parse(editRedirectRequest));
}
export function editRedirectRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditRedirectRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditRedirectRequest' from JSON`);
}
/** @internal */
export const EditRedirectVersion$inboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
/** @internal */
export const EditRedirectVersion$outboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
export function editRedirectVersionToJSON(editRedirectVersion) {
    return JSON.stringify(EditRedirectVersion$outboundSchema.parse(editRedirectVersion));
}
export function editRedirectVersionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditRedirectVersion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditRedirectVersion' from JSON`);
}
/** @internal */
export const EditRedirectResponseBody$inboundSchema = z.object({
    alias: z.nullable(z.string()),
    version: z.lazy(() => EditRedirectVersion$inboundSchema),
});
/** @internal */
export const EditRedirectResponseBody$outboundSchema = z.object({
    alias: z.nullable(z.string()),
    version: z.lazy(() => EditRedirectVersion$outboundSchema),
});
export function editRedirectResponseBodyToJSON(editRedirectResponseBody) {
    return JSON.stringify(EditRedirectResponseBody$outboundSchema.parse(editRedirectResponseBody));
}
export function editRedirectResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => EditRedirectResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'EditRedirectResponseBody' from JSON`);
}
