// @vercel/sdk/esm/models/getfirewallconfigop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSdAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponseAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponse200Action = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsXssAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSqliAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSfAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsJavaAction = {
    Deny: "deny",
    Log: "log",
};
export const GetFirewallConfigRulesType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Protocol: "protocol",
    Region: "region",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
    ServerAction: "server_action",
    BotName: "bot_name",
    BotCategory: "bot_category",
};
export const GetFirewallConfigRulesOp = {
    Re: "re",
    Eq: "eq",
    Ex: "ex",
    Inc: "inc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
    Nex: "nex",
    Ninc: "ninc",
    Neq: "neq",
};
export const GetFirewallConfigRulesSecurityAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const GetFirewallConfigRulesAlgo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const GetFirewallConfigRulesSecurityResponse200Action = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    RateLimit: "rate_limit",
};
export const RulesType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Protocol: "protocol",
    Region: "region",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
    ServerAction: "server_action",
    BotName: "bot_name",
    BotCategory: "bot_category",
};
export const RulesOp = {
    Re: "re",
    Eq: "eq",
    Ex: "ex",
    Inc: "inc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
    Nex: "nex",
    Ninc: "ninc",
    Neq: "neq",
};
export const GetFirewallConfigRulesSecurityResponseAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const RulesAlgo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const GetFirewallConfigRulesSecurityResponse200ApplicationJSONAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    RateLimit: "rate_limit",
};
export const GetFirewallConfigAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    Bypass: "bypass",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAiBotsAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesOwaspAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
/** @internal */
export const GetFirewallConfigRequest$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    configVersion: z.string(),
});
/** @internal */
export const GetFirewallConfigRequest$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    configVersion: z.string(),
});
export function getFirewallConfigRequestToJSON(getFirewallConfigRequest) {
    return JSON.stringify(GetFirewallConfigRequest$outboundSchema.parse(getFirewallConfigRequest));
}
export function getFirewallConfigRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRequest' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSdAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSdAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSdAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSdAction$inboundSchema;
/** @internal */
export const GetFirewallConfigSd$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSdAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigSd$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSdAction$outboundSchema,
});
export function getFirewallConfigSdToJSON(getFirewallConfigSd) {
    return JSON.stringify(GetFirewallConfigSd$outboundSchema.parse(getFirewallConfigSd));
}
export function getFirewallConfigSdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigSd$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigSd' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityAction);
/** @internal */
export const GetFirewallConfigSecurityAction$outboundSchema = GetFirewallConfigSecurityAction$inboundSchema;
/** @internal */
export const GetFirewallConfigMa$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigMa$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityAction$outboundSchema,
});
export function getFirewallConfigMaToJSON(getFirewallConfigMa) {
    return JSON.stringify(GetFirewallConfigMa$outboundSchema.parse(getFirewallConfigMa));
}
export function getFirewallConfigMaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigMa$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigMa' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponseAction$inboundSchema = z
    .nativeEnum(GetFirewallConfigSecurityResponseAction);
/** @internal */
export const GetFirewallConfigSecurityResponseAction$outboundSchema = GetFirewallConfigSecurityResponseAction$inboundSchema;
/** @internal */
export const GetFirewallConfigLfi$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponseAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigLfi$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponseAction$outboundSchema,
});
export function getFirewallConfigLfiToJSON(getFirewallConfigLfi) {
    return JSON.stringify(GetFirewallConfigLfi$outboundSchema.parse(getFirewallConfigLfi));
}
export function getFirewallConfigLfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigLfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigLfi' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200Action$inboundSchema = z
    .nativeEnum(GetFirewallConfigSecurityResponse200Action);
/** @internal */
export const GetFirewallConfigSecurityResponse200Action$outboundSchema = GetFirewallConfigSecurityResponse200Action$inboundSchema;
/** @internal */
export const GetFirewallConfigRfi$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200Action$inboundSchema,
});
/** @internal */
export const GetFirewallConfigRfi$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200Action$outboundSchema,
});
export function getFirewallConfigRfiToJSON(getFirewallConfigRfi) {
    return JSON.stringify(GetFirewallConfigRfi$outboundSchema.parse(getFirewallConfigRfi));
}
export function getFirewallConfigRfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRfi' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema;
/** @internal */
export const GetFirewallConfigRce$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigRce$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONAction$outboundSchema,
});
export function getFirewallConfigRceToJSON(getFirewallConfigRce) {
    return JSON.stringify(GetFirewallConfigRce$outboundSchema.parse(getFirewallConfigRce));
}
export function getFirewallConfigRceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRce$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRce' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema;
/** @internal */
export const GetFirewallConfigPhp$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigPhp$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema,
});
export function getFirewallConfigPhpToJSON(getFirewallConfigPhp) {
    return JSON.stringify(GetFirewallConfigPhp$outboundSchema.parse(getFirewallConfigPhp));
}
export function getFirewallConfigPhpFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigPhp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigPhp' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsAction$inboundSchema;
/** @internal */
export const GetFirewallConfigGen$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigGen$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsAction$outboundSchema,
});
export function getFirewallConfigGenToJSON(getFirewallConfigGen) {
    return JSON.stringify(GetFirewallConfigGen$outboundSchema.parse(getFirewallConfigGen));
}
export function getFirewallConfigGenFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigGen$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigGen' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsXssAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsXssAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsXssAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsXssAction$inboundSchema;
/** @internal */
export const GetFirewallConfigXss$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsXssAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigXss$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsXssAction$outboundSchema,
});
export function getFirewallConfigXssToJSON(getFirewallConfigXss) {
    return JSON.stringify(GetFirewallConfigXss$outboundSchema.parse(getFirewallConfigXss));
}
export function getFirewallConfigXssFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigXss$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigXss' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSqliAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSqliAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSqliAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSqliAction$inboundSchema;
/** @internal */
export const GetFirewallConfigSqli$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSqliAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigSqli$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSqliAction$outboundSchema,
});
export function getFirewallConfigSqliToJSON(getFirewallConfigSqli) {
    return JSON.stringify(GetFirewallConfigSqli$outboundSchema.parse(getFirewallConfigSqli));
}
export function getFirewallConfigSqliFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigSqli$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigSqli' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSfAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSfAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSfAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSfAction$inboundSchema;
/** @internal */
export const GetFirewallConfigSf$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSfAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigSf$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsSfAction$outboundSchema,
});
export function getFirewallConfigSfToJSON(getFirewallConfigSf) {
    return JSON.stringify(GetFirewallConfigSf$outboundSchema.parse(getFirewallConfigSf));
}
export function getFirewallConfigSfFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigSf$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigSf' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsJavaAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsJavaAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsJavaAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsJavaAction$inboundSchema;
/** @internal */
export const GetFirewallConfigJava$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsJavaAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigJava$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyCrsJavaAction$outboundSchema,
});
export function getFirewallConfigJavaToJSON(getFirewallConfigJava) {
    return JSON.stringify(GetFirewallConfigJava$outboundSchema.parse(getFirewallConfigJava));
}
export function getFirewallConfigJavaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigJava$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigJava' from JSON`);
}
/** @internal */
export const GetFirewallConfigCrs$inboundSchema = z.object({
    sd: z.lazy(() => GetFirewallConfigSd$inboundSchema),
    ma: z.lazy(() => GetFirewallConfigMa$inboundSchema),
    lfi: z.lazy(() => GetFirewallConfigLfi$inboundSchema),
    rfi: z.lazy(() => GetFirewallConfigRfi$inboundSchema),
    rce: z.lazy(() => GetFirewallConfigRce$inboundSchema),
    php: z.lazy(() => GetFirewallConfigPhp$inboundSchema),
    gen: z.lazy(() => GetFirewallConfigGen$inboundSchema),
    xss: z.lazy(() => GetFirewallConfigXss$inboundSchema),
    sqli: z.lazy(() => GetFirewallConfigSqli$inboundSchema),
    sf: z.lazy(() => GetFirewallConfigSf$inboundSchema),
    java: z.lazy(() => GetFirewallConfigJava$inboundSchema),
});
/** @internal */
export const GetFirewallConfigCrs$outboundSchema = z.object({
    sd: z.lazy(() => GetFirewallConfigSd$outboundSchema),
    ma: z.lazy(() => GetFirewallConfigMa$outboundSchema),
    lfi: z.lazy(() => GetFirewallConfigLfi$outboundSchema),
    rfi: z.lazy(() => GetFirewallConfigRfi$outboundSchema),
    rce: z.lazy(() => GetFirewallConfigRce$outboundSchema),
    php: z.lazy(() => GetFirewallConfigPhp$outboundSchema),
    gen: z.lazy(() => GetFirewallConfigGen$outboundSchema),
    xss: z.lazy(() => GetFirewallConfigXss$outboundSchema),
    sqli: z.lazy(() => GetFirewallConfigSqli$outboundSchema),
    sf: z.lazy(() => GetFirewallConfigSf$outboundSchema),
    java: z.lazy(() => GetFirewallConfigJava$outboundSchema),
});
export function getFirewallConfigCrsToJSON(getFirewallConfigCrs) {
    return JSON.stringify(GetFirewallConfigCrs$outboundSchema.parse(getFirewallConfigCrs));
}
export function getFirewallConfigCrsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigCrs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigCrs' from JSON`);
}
/** @internal */
export const GetFirewallConfigRulesType$inboundSchema = z.nativeEnum(GetFirewallConfigRulesType);
/** @internal */
export const GetFirewallConfigRulesType$outboundSchema = GetFirewallConfigRulesType$inboundSchema;
/** @internal */
export const GetFirewallConfigRulesOp$inboundSchema = z.nativeEnum(GetFirewallConfigRulesOp);
/** @internal */
export const GetFirewallConfigRulesOp$outboundSchema = GetFirewallConfigRulesOp$inboundSchema;
/** @internal */
export const GetFirewallConfigRulesValue$inboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
/** @internal */
export const GetFirewallConfigRulesValue$outboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
export function getFirewallConfigRulesValueToJSON(getFirewallConfigRulesValue) {
    return JSON.stringify(GetFirewallConfigRulesValue$outboundSchema.parse(getFirewallConfigRulesValue));
}
export function getFirewallConfigRulesValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRulesValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRulesValue' from JSON`);
}
/** @internal */
export const GetFirewallConfigRulesConditions$inboundSchema = z.object({
    type: GetFirewallConfigRulesType$inboundSchema,
    op: GetFirewallConfigRulesOp$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
/** @internal */
export const GetFirewallConfigRulesConditions$outboundSchema = z.object({
    type: GetFirewallConfigRulesType$outboundSchema,
    op: GetFirewallConfigRulesOp$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
export function getFirewallConfigRulesConditionsToJSON(getFirewallConfigRulesConditions) {
    return JSON.stringify(GetFirewallConfigRulesConditions$outboundSchema.parse(getFirewallConfigRulesConditions));
}
export function getFirewallConfigRulesConditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRulesConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRulesConditions' from JSON`);
}
/** @internal */
export const GetFirewallConfigRulesConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => GetFirewallConfigRulesConditions$inboundSchema)),
});
/** @internal */
export const GetFirewallConfigRulesConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => GetFirewallConfigRulesConditions$outboundSchema)),
});
export function getFirewallConfigRulesConditionGroupToJSON(getFirewallConfigRulesConditionGroup) {
    return JSON.stringify(GetFirewallConfigRulesConditionGroup$outboundSchema.parse(getFirewallConfigRulesConditionGroup));
}
export function getFirewallConfigRulesConditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRulesConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRulesConditionGroup' from JSON`);
}
/** @internal */
export const GetFirewallConfigRulesSecurityAction$inboundSchema = z.nativeEnum(GetFirewallConfigRulesSecurityAction);
/** @internal */
export const GetFirewallConfigRulesSecurityAction$outboundSchema = GetFirewallConfigRulesSecurityAction$inboundSchema;
/** @internal */
export const GetFirewallConfigRulesAlgo$inboundSchema = z.nativeEnum(GetFirewallConfigRulesAlgo);
/** @internal */
export const GetFirewallConfigRulesAlgo$outboundSchema = GetFirewallConfigRulesAlgo$inboundSchema;
/** @internal */
export const GetFirewallConfigRulesSecurityResponse200Action$inboundSchema = z
    .nativeEnum(GetFirewallConfigRulesSecurityResponse200Action);
/** @internal */
export const GetFirewallConfigRulesSecurityResponse200Action$outboundSchema = GetFirewallConfigRulesSecurityResponse200Action$inboundSchema;
/** @internal */
export const GetFirewallConfigRulesRateLimit$inboundSchema = z.object({
    algo: GetFirewallConfigRulesAlgo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(GetFirewallConfigRulesSecurityResponse200Action$inboundSchema).optional(),
});
/** @internal */
export const GetFirewallConfigRulesRateLimit$outboundSchema = z.object({
    algo: GetFirewallConfigRulesAlgo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(GetFirewallConfigRulesSecurityResponse200Action$outboundSchema).optional(),
});
export function getFirewallConfigRulesRateLimitToJSON(getFirewallConfigRulesRateLimit) {
    return JSON.stringify(GetFirewallConfigRulesRateLimit$outboundSchema.parse(getFirewallConfigRulesRateLimit));
}
export function getFirewallConfigRulesRateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRulesRateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRulesRateLimit' from JSON`);
}
/** @internal */
export const GetFirewallConfigRulesRedirect$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const GetFirewallConfigRulesRedirect$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
export function getFirewallConfigRulesRedirectToJSON(getFirewallConfigRulesRedirect) {
    return JSON.stringify(GetFirewallConfigRulesRedirect$outboundSchema.parse(getFirewallConfigRulesRedirect));
}
export function getFirewallConfigRulesRedirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRulesRedirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRulesRedirect' from JSON`);
}
/** @internal */
export const GetFirewallConfigRulesMitigate$inboundSchema = z.object({
    action: GetFirewallConfigRulesSecurityAction$inboundSchema,
    rateLimit: z.nullable(z.lazy(() => GetFirewallConfigRulesRateLimit$inboundSchema)).optional(),
    redirect: z.nullable(z.lazy(() => GetFirewallConfigRulesRedirect$inboundSchema)).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const GetFirewallConfigRulesMitigate$outboundSchema = z.object({
    action: GetFirewallConfigRulesSecurityAction$outboundSchema,
    rateLimit: z.nullable(z.lazy(() => GetFirewallConfigRulesRateLimit$outboundSchema)).optional(),
    redirect: z.nullable(z.lazy(() => GetFirewallConfigRulesRedirect$outboundSchema)).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
export function getFirewallConfigRulesMitigateToJSON(getFirewallConfigRulesMitigate) {
    return JSON.stringify(GetFirewallConfigRulesMitigate$outboundSchema.parse(getFirewallConfigRulesMitigate));
}
export function getFirewallConfigRulesMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRulesMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRulesMitigate' from JSON`);
}
/** @internal */
export const GetFirewallConfigRulesAction$inboundSchema = z.object({
    mitigate: z.lazy(() => GetFirewallConfigRulesMitigate$inboundSchema)
        .optional(),
});
/** @internal */
export const GetFirewallConfigRulesAction$outboundSchema = z.object({
    mitigate: z.lazy(() => GetFirewallConfigRulesMitigate$outboundSchema)
        .optional(),
});
export function getFirewallConfigRulesActionToJSON(getFirewallConfigRulesAction) {
    return JSON.stringify(GetFirewallConfigRulesAction$outboundSchema.parse(getFirewallConfigRulesAction));
}
export function getFirewallConfigRulesActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRulesAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRulesAction' from JSON`);
}
/** @internal */
export const Rules2$inboundSchema = z
    .object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => GetFirewallConfigRulesConditionGroup$inboundSchema)),
    action: z.lazy(() => GetFirewallConfigRulesAction$inboundSchema),
    valid: z.boolean(),
    validationErrors: z.array(z.string()),
});
/** @internal */
export const Rules2$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => GetFirewallConfigRulesConditionGroup$outboundSchema)),
    action: z.lazy(() => GetFirewallConfigRulesAction$outboundSchema),
    valid: z.boolean(),
    validationErrors: z.array(z.string()),
});
export function rules2ToJSON(rules2) {
    return JSON.stringify(Rules2$outboundSchema.parse(rules2));
}
export function rules2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Rules2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Rules2' from JSON`);
}
/** @internal */
export const RulesType$inboundSchema = z
    .nativeEnum(RulesType);
/** @internal */
export const RulesType$outboundSchema = RulesType$inboundSchema;
/** @internal */
export const RulesOp$inboundSchema = z
    .nativeEnum(RulesOp);
/** @internal */
export const RulesOp$outboundSchema = RulesOp$inboundSchema;
/** @internal */
export const RulesValue$inboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
/** @internal */
export const RulesValue$outboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
export function rulesValueToJSON(rulesValue) {
    return JSON.stringify(RulesValue$outboundSchema.parse(rulesValue));
}
export function rulesValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RulesValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RulesValue' from JSON`);
}
/** @internal */
export const RulesConditions$inboundSchema = z.object({
    type: RulesType$inboundSchema,
    op: RulesOp$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
/** @internal */
export const RulesConditions$outboundSchema = z.object({
    type: RulesType$outboundSchema,
    op: RulesOp$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
export function rulesConditionsToJSON(rulesConditions) {
    return JSON.stringify(RulesConditions$outboundSchema.parse(rulesConditions));
}
export function rulesConditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RulesConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RulesConditions' from JSON`);
}
/** @internal */
export const RulesConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => RulesConditions$inboundSchema)),
});
/** @internal */
export const RulesConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => RulesConditions$outboundSchema)),
});
export function rulesConditionGroupToJSON(rulesConditionGroup) {
    return JSON.stringify(RulesConditionGroup$outboundSchema.parse(rulesConditionGroup));
}
export function rulesConditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RulesConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RulesConditionGroup' from JSON`);
}
/** @internal */
export const GetFirewallConfigRulesSecurityResponseAction$inboundSchema = z
    .nativeEnum(GetFirewallConfigRulesSecurityResponseAction);
/** @internal */
export const GetFirewallConfigRulesSecurityResponseAction$outboundSchema = GetFirewallConfigRulesSecurityResponseAction$inboundSchema;
/** @internal */
export const RulesAlgo$inboundSchema = z
    .nativeEnum(RulesAlgo);
/** @internal */
export const RulesAlgo$outboundSchema = RulesAlgo$inboundSchema;
/** @internal */
export const GetFirewallConfigRulesSecurityResponse200ApplicationJSONAction$inboundSchema = z.nativeEnum(GetFirewallConfigRulesSecurityResponse200ApplicationJSONAction);
/** @internal */
export const GetFirewallConfigRulesSecurityResponse200ApplicationJSONAction$outboundSchema = GetFirewallConfigRulesSecurityResponse200ApplicationJSONAction$inboundSchema;
/** @internal */
export const RulesRateLimit$inboundSchema = z.object({
    algo: RulesAlgo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(GetFirewallConfigRulesSecurityResponse200ApplicationJSONAction$inboundSchema).optional(),
});
/** @internal */
export const RulesRateLimit$outboundSchema = z.object({
    algo: RulesAlgo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(GetFirewallConfigRulesSecurityResponse200ApplicationJSONAction$outboundSchema).optional(),
});
export function rulesRateLimitToJSON(rulesRateLimit) {
    return JSON.stringify(RulesRateLimit$outboundSchema.parse(rulesRateLimit));
}
export function rulesRateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RulesRateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RulesRateLimit' from JSON`);
}
/** @internal */
export const RulesRedirect$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const RulesRedirect$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
export function rulesRedirectToJSON(rulesRedirect) {
    return JSON.stringify(RulesRedirect$outboundSchema.parse(rulesRedirect));
}
export function rulesRedirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RulesRedirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RulesRedirect' from JSON`);
}
/** @internal */
export const RulesMitigate$inboundSchema = z.object({
    action: GetFirewallConfigRulesSecurityResponseAction$inboundSchema,
    rateLimit: z.nullable(z.lazy(() => RulesRateLimit$inboundSchema)).optional(),
    redirect: z.nullable(z.lazy(() => RulesRedirect$inboundSchema)).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const RulesMitigate$outboundSchema = z.object({
    action: GetFirewallConfigRulesSecurityResponseAction$outboundSchema,
    rateLimit: z.nullable(z.lazy(() => RulesRateLimit$outboundSchema)).optional(),
    redirect: z.nullable(z.lazy(() => RulesRedirect$outboundSchema)).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
export function rulesMitigateToJSON(rulesMitigate) {
    return JSON.stringify(RulesMitigate$outboundSchema.parse(rulesMitigate));
}
export function rulesMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RulesMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RulesMitigate' from JSON`);
}
/** @internal */
export const RulesAction$inboundSchema = z.object({
    mitigate: z.lazy(() => RulesMitigate$inboundSchema).optional(),
});
/** @internal */
export const RulesAction$outboundSchema = z.object({
    mitigate: z.lazy(() => RulesMitigate$outboundSchema).optional(),
});
export function rulesActionToJSON(rulesAction) {
    return JSON.stringify(RulesAction$outboundSchema.parse(rulesAction));
}
export function rulesActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RulesAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RulesAction' from JSON`);
}
/** @internal */
export const Rules1$inboundSchema = z
    .object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => RulesConditionGroup$inboundSchema)),
    action: z.lazy(() => RulesAction$inboundSchema),
    valid: z.boolean(),
    validationErrors: z.nullable(z.any()).optional(),
});
/** @internal */
export const Rules1$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => RulesConditionGroup$outboundSchema)),
    action: z.lazy(() => RulesAction$outboundSchema),
    valid: z.boolean(),
    validationErrors: z.nullable(z.any()).optional(),
});
export function rules1ToJSON(rules1) {
    return JSON.stringify(Rules1$outboundSchema.parse(rules1));
}
export function rules1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Rules1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Rules1' from JSON`);
}
/** @internal */
export const GetFirewallConfigRules$inboundSchema = z.union([
    z.lazy(() => Rules1$inboundSchema),
    z.lazy(() => Rules2$inboundSchema),
]);
/** @internal */
export const GetFirewallConfigRules$outboundSchema = z.union([
    z.lazy(() => Rules1$outboundSchema),
    z.lazy(() => Rules2$outboundSchema),
]);
export function getFirewallConfigRulesToJSON(getFirewallConfigRules) {
    return JSON.stringify(GetFirewallConfigRules$outboundSchema.parse(getFirewallConfigRules));
}
export function getFirewallConfigRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigRules' from JSON`);
}
/** @internal */
export const GetFirewallConfigAction$inboundSchema = z.nativeEnum(GetFirewallConfigAction);
/** @internal */
export const GetFirewallConfigAction$outboundSchema = GetFirewallConfigAction$inboundSchema;
/** @internal */
export const GetFirewallConfigIps$inboundSchema = z.object({
    id: z.string(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: GetFirewallConfigAction$inboundSchema,
});
/** @internal */
export const GetFirewallConfigIps$outboundSchema = z.object({
    id: z.string(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: GetFirewallConfigAction$outboundSchema,
});
export function getFirewallConfigIpsToJSON(getFirewallConfigIps) {
    return JSON.stringify(GetFirewallConfigIps$outboundSchema.parse(getFirewallConfigIps));
}
export function getFirewallConfigIpsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigIps$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigIps' from JSON`);
}
/** @internal */
export const Changes$inboundSchema = z.object({});
/** @internal */
export const Changes$outboundSchema = z.object({});
export function changesToJSON(changes) {
    return JSON.stringify(Changes$outboundSchema.parse(changes));
}
export function changesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Changes$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Changes' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAction$inboundSchema;
/** @internal */
export const BotProtection$inboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const BotProtection$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
export function botProtectionToJSON(botProtection) {
    return JSON.stringify(BotProtection$outboundSchema.parse(botProtection));
}
export function botProtectionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => BotProtection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BotProtection' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAiBotsAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAiBotsAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAiBotsAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAiBotsAction$inboundSchema;
/** @internal */
export const AiBots$inboundSchema = z
    .object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAiBotsAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const AiBots$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesAiBotsAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
export function aiBotsToJSON(aiBots) {
    return JSON.stringify(AiBots$outboundSchema.parse(aiBots));
}
export function aiBotsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => AiBots$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'AiBots' from JSON`);
}
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesOwaspAction$inboundSchema = z.nativeEnum(GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesOwaspAction);
/** @internal */
export const GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesOwaspAction$outboundSchema = GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesOwaspAction$inboundSchema;
/** @internal */
export const Owasp$inboundSchema = z
    .object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesOwaspAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const Owasp$outboundSchema = z.object({
    active: z.boolean(),
    action: GetFirewallConfigSecurityResponse200ApplicationJSONResponseBodyManagedRulesOwaspAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
export function owaspToJSON(owasp) {
    return JSON.stringify(Owasp$outboundSchema.parse(owasp));
}
export function owaspFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Owasp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Owasp' from JSON`);
}
/** @internal */
export const GetFirewallConfigManagedRules$inboundSchema = z.object({
    bot_protection: z.lazy(() => BotProtection$inboundSchema).optional(),
    ai_bots: z.lazy(() => AiBots$inboundSchema).optional(),
    owasp: z.lazy(() => Owasp$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "bot_protection": "botProtection",
        "ai_bots": "aiBots",
    });
});
/** @internal */
export const GetFirewallConfigManagedRules$outboundSchema = z.object({
    botProtection: z.lazy(() => BotProtection$outboundSchema).optional(),
    aiBots: z.lazy(() => AiBots$outboundSchema).optional(),
    owasp: z.lazy(() => Owasp$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        botProtection: "bot_protection",
        aiBots: "ai_bots",
    });
});
export function getFirewallConfigManagedRulesToJSON(getFirewallConfigManagedRules) {
    return JSON.stringify(GetFirewallConfigManagedRules$outboundSchema.parse(getFirewallConfigManagedRules));
}
export function getFirewallConfigManagedRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigManagedRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigManagedRules' from JSON`);
}
/** @internal */
export const GetFirewallConfigResponseBody$inboundSchema = z.object({
    ownerId: z.string(),
    projectKey: z.string(),
    id: z.string(),
    version: z.number(),
    updatedAt: z.string(),
    firewallEnabled: z.boolean(),
    crs: z.lazy(() => GetFirewallConfigCrs$inboundSchema),
    rules: z.array(z.union([
        z.lazy(() => Rules1$inboundSchema),
        z.lazy(() => Rules2$inboundSchema),
    ])),
    ips: z.array(z.lazy(() => GetFirewallConfigIps$inboundSchema)),
    changes: z.array(z.lazy(() => Changes$inboundSchema)),
    managedRules: z.lazy(() => GetFirewallConfigManagedRules$inboundSchema)
        .optional(),
    botIdEnabled: z.boolean().optional(),
});
/** @internal */
export const GetFirewallConfigResponseBody$outboundSchema = z.object({
    ownerId: z.string(),
    projectKey: z.string(),
    id: z.string(),
    version: z.number(),
    updatedAt: z.string(),
    firewallEnabled: z.boolean(),
    crs: z.lazy(() => GetFirewallConfigCrs$outboundSchema),
    rules: z.array(z.union([
        z.lazy(() => Rules1$outboundSchema),
        z.lazy(() => Rules2$outboundSchema),
    ])),
    ips: z.array(z.lazy(() => GetFirewallConfigIps$outboundSchema)),
    changes: z.array(z.lazy(() => Changes$outboundSchema)),
    managedRules: z.lazy(() => GetFirewallConfigManagedRules$outboundSchema)
        .optional(),
    botIdEnabled: z.boolean().optional(),
});
export function getFirewallConfigResponseBodyToJSON(getFirewallConfigResponseBody) {
    return JSON.stringify(GetFirewallConfigResponseBody$outboundSchema.parse(getFirewallConfigResponseBody));
}
export function getFirewallConfigResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetFirewallConfigResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetFirewallConfigResponseBody' from JSON`);
}
