// @vercel/sdk/esm/models/getredirectsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const QueryParam2 = {
    Only: "only",
};
export const SortBy = {
    Source: "source",
    Destination: "destination",
    StatusCode: "statusCode",
};
export const SortOrder = {
    Asc: "asc",
    Desc: "desc",
};
/** @internal */
export const QueryParam2$inboundSchema = z
    .nativeEnum(QueryParam2);
/** @internal */
export const QueryParam2$outboundSchema = QueryParam2$inboundSchema;
/** @internal */
export const Diff$inboundSchema = z
    .union([z.boolean(), QueryParam2$inboundSchema]);
/** @internal */
export const Diff$outboundSchema = z.union([z.boolean(), QueryParam2$outboundSchema]);
export function diffToJSON(diff) {
    return JSON.stringify(Diff$outboundSchema.parse(diff));
}
export function diffFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Diff$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Diff' from JSON`);
}
/** @internal */
export const SortBy$inboundSchema = z
    .nativeEnum(SortBy);
/** @internal */
export const SortBy$outboundSchema = SortBy$inboundSchema;
/** @internal */
export const SortOrder$inboundSchema = z
    .nativeEnum(SortOrder);
/** @internal */
export const SortOrder$outboundSchema = SortOrder$inboundSchema;
/** @internal */
export const GetRedirectsRequest$inboundSchema = z.object({
    projectId: z.string(),
    versionId: z.string().optional(),
    q: z.string().optional(),
    diff: z.union([z.boolean(), QueryParam2$inboundSchema]).optional(),
    page: z.number().int().optional(),
    per_page: z.number().int().optional(),
    sort_by: SortBy$inboundSchema.optional(),
    sort_order: SortOrder$inboundSchema.optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        "per_page": "perPage",
        "sort_by": "sortBy",
        "sort_order": "sortOrder",
    });
});
/** @internal */
export const GetRedirectsRequest$outboundSchema = z.object({
    projectId: z.string(),
    versionId: z.string().optional(),
    q: z.string().optional(),
    diff: z.union([z.boolean(), QueryParam2$outboundSchema]).optional(),
    page: z.number().int().optional(),
    perPage: z.number().int().optional(),
    sortBy: SortBy$outboundSchema.optional(),
    sortOrder: SortOrder$outboundSchema.optional(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
}).transform((v) => {
    return remap$(v, {
        perPage: "per_page",
        sortBy: "sort_by",
        sortOrder: "sort_order",
    });
});
export function getRedirectsRequestToJSON(getRedirectsRequest) {
    return JSON.stringify(GetRedirectsRequest$outboundSchema.parse(getRedirectsRequest));
}
export function getRedirectsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRedirectsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRedirectsRequest' from JSON`);
}
/** @internal */
export const ResponseBodyRedirects$inboundSchema = z.object({
    statusCode: z.number().optional(),
    permanent: z.boolean().optional(),
    sensitive: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    query: z.boolean().optional(),
    destination: z.string(),
    source: z.string(),
});
/** @internal */
export const ResponseBodyRedirects$outboundSchema = z.object({
    statusCode: z.number().optional(),
    permanent: z.boolean().optional(),
    sensitive: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    query: z.boolean().optional(),
    destination: z.string(),
    source: z.string(),
});
export function responseBodyRedirectsToJSON(responseBodyRedirects) {
    return JSON.stringify(ResponseBodyRedirects$outboundSchema.parse(responseBodyRedirects));
}
export function responseBodyRedirectsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseBodyRedirects$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseBodyRedirects' from JSON`);
}
/** @internal */
export const ResponseBodyVersion$inboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
/** @internal */
export const ResponseBodyVersion$outboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
export function responseBodyVersionToJSON(responseBodyVersion) {
    return JSON.stringify(ResponseBodyVersion$outboundSchema.parse(responseBodyVersion));
}
export function responseBodyVersionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ResponseBodyVersion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ResponseBodyVersion' from JSON`);
}
/** @internal */
export const GetRedirectsResponseBodyPagination$inboundSchema = z.object({
    page: z.number(),
    per_page: z.number(),
    numPages: z.number(),
}).transform((v) => {
    return remap$(v, {
        "per_page": "perPage",
    });
});
/** @internal */
export const GetRedirectsResponseBodyPagination$outboundSchema = z.object({
    page: z.number(),
    perPage: z.number(),
    numPages: z.number(),
}).transform((v) => {
    return remap$(v, {
        perPage: "per_page",
    });
});
export function getRedirectsResponseBodyPaginationToJSON(getRedirectsResponseBodyPagination) {
    return JSON.stringify(GetRedirectsResponseBodyPagination$outboundSchema.parse(getRedirectsResponseBodyPagination));
}
export function getRedirectsResponseBodyPaginationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRedirectsResponseBodyPagination$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRedirectsResponseBodyPagination' from JSON`);
}
/** @internal */
export const GetRedirectsResponseBody2$inboundSchema = z.object({
    redirects: z.array(z.lazy(() => ResponseBodyRedirects$inboundSchema)),
    version: z.lazy(() => ResponseBodyVersion$inboundSchema).optional(),
    pagination: z.lazy(() => GetRedirectsResponseBodyPagination$inboundSchema),
});
/** @internal */
export const GetRedirectsResponseBody2$outboundSchema = z.object({
    redirects: z.array(z.lazy(() => ResponseBodyRedirects$outboundSchema)),
    version: z.lazy(() => ResponseBodyVersion$outboundSchema).optional(),
    pagination: z.lazy(() => GetRedirectsResponseBodyPagination$outboundSchema),
});
export function getRedirectsResponseBody2ToJSON(getRedirectsResponseBody2) {
    return JSON.stringify(GetRedirectsResponseBody2$outboundSchema.parse(getRedirectsResponseBody2));
}
export function getRedirectsResponseBody2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRedirectsResponseBody2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRedirectsResponseBody2' from JSON`);
}
/** @internal */
export const GetRedirectsResponseBody$inboundSchema = z.union([
    z.lazy(() => GetRedirectsResponseBody2$inboundSchema),
    z.record(z.any()),
]);
/** @internal */
export const GetRedirectsResponseBody$outboundSchema = z.union([
    z.lazy(() => GetRedirectsResponseBody2$outboundSchema),
    z.record(z.any()),
]);
export function getRedirectsResponseBodyToJSON(getRedirectsResponseBody) {
    return JSON.stringify(GetRedirectsResponseBody$outboundSchema.parse(getRedirectsResponseBody));
}
export function getRedirectsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetRedirectsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetRedirectsResponseBody' from JSON`);
}
