// @vercel/sdk/esm/models/getversionsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetVersionsRequest$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const GetVersionsRequest$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
export function getVersionsRequestToJSON(getVersionsRequest) {
    return JSON.stringify(GetVersionsRequest$outboundSchema.parse(getVersionsRequest));
}
export function getVersionsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetVersionsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetVersionsRequest' from JSON`);
}
/** @internal */
export const Versions$inboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
/** @internal */
export const Versions$outboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
export function versionsToJSON(versions) {
    return JSON.stringify(Versions$outboundSchema.parse(versions));
}
export function versionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Versions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Versions' from JSON`);
}
/** @internal */
export const GetVersionsResponseBody$inboundSchema = z.object({
    versions: z.array(z.lazy(() => Versions$inboundSchema)),
});
/** @internal */
export const GetVersionsResponseBody$outboundSchema = z.object({
    versions: z.array(z.lazy(() => Versions$outboundSchema)),
});
export function getVersionsResponseBodyToJSON(getVersionsResponseBody) {
    return JSON.stringify(GetVersionsResponseBody$outboundSchema.parse(getVersionsResponseBody));
}
export function getVersionsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetVersionsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetVersionsResponseBody' from JSON`);
}
