// @vercel/sdk/esm/models/moveprojectdomainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/**
 * Status code for domain redirect
 */
export const MoveProjectDomainRedirectStatusCode = {
    ThreeHundredAndOne: 301,
    ThreeHundredAndTwo: 302,
    ThreeHundredAndSeven: 307,
    ThreeHundredAndEight: 308,
};
/** @internal */
export const MoveProjectDomainRedirectStatusCode$inboundSchema = z.nativeEnum(MoveProjectDomainRedirectStatusCode);
/** @internal */
export const MoveProjectDomainRedirectStatusCode$outboundSchema = MoveProjectDomainRedirectStatusCode$inboundSchema;
/** @internal */
export const MoveProjectDomainRequestBody$inboundSchema = z.object({
    projectId: z.string(),
    gitBranch: z.nullable(z.string()).optional(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(MoveProjectDomainRedirectStatusCode$inboundSchema).optional(),
});
/** @internal */
export const MoveProjectDomainRequestBody$outboundSchema = z.object({
    projectId: z.string(),
    gitBranch: z.nullable(z.string()).optional(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(MoveProjectDomainRedirectStatusCode$outboundSchema).optional(),
});
export function moveProjectDomainRequestBodyToJSON(moveProjectDomainRequestBody) {
    return JSON.stringify(MoveProjectDomainRequestBody$outboundSchema.parse(moveProjectDomainRequestBody));
}
export function moveProjectDomainRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MoveProjectDomainRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MoveProjectDomainRequestBody' from JSON`);
}
/** @internal */
export const MoveProjectDomainRequest$inboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => MoveProjectDomainRequestBody$inboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const MoveProjectDomainRequest$outboundSchema = z.object({
    idOrName: z.string(),
    domain: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => MoveProjectDomainRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function moveProjectDomainRequestToJSON(moveProjectDomainRequest) {
    return JSON.stringify(MoveProjectDomainRequest$outboundSchema.parse(moveProjectDomainRequest));
}
export function moveProjectDomainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MoveProjectDomainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MoveProjectDomainRequest' from JSON`);
}
/** @internal */
export const MoveProjectDomainVerification$inboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
/** @internal */
export const MoveProjectDomainVerification$outboundSchema = z.object({
    type: z.string(),
    domain: z.string(),
    value: z.string(),
    reason: z.string(),
});
export function moveProjectDomainVerificationToJSON(moveProjectDomainVerification) {
    return JSON.stringify(MoveProjectDomainVerification$outboundSchema.parse(moveProjectDomainVerification));
}
export function moveProjectDomainVerificationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MoveProjectDomainVerification$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MoveProjectDomainVerification' from JSON`);
}
/** @internal */
export const MoveProjectDomainResponseBody$inboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => MoveProjectDomainVerification$inboundSchema)).optional(),
});
/** @internal */
export const MoveProjectDomainResponseBody$outboundSchema = z.object({
    name: z.string(),
    apexName: z.string(),
    projectId: z.string(),
    redirect: z.nullable(z.string()).optional(),
    redirectStatusCode: z.nullable(z.number()).optional(),
    gitBranch: z.nullable(z.string()).optional(),
    customEnvironmentId: z.nullable(z.string()).optional(),
    updatedAt: z.number().optional(),
    createdAt: z.number().optional(),
    verified: z.boolean(),
    verification: z.array(z.lazy(() => MoveProjectDomainVerification$outboundSchema)).optional(),
});
export function moveProjectDomainResponseBodyToJSON(moveProjectDomainResponseBody) {
    return JSON.stringify(MoveProjectDomainResponseBody$outboundSchema.parse(moveProjectDomainResponseBody));
}
export function moveProjectDomainResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => MoveProjectDomainResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'MoveProjectDomainResponseBody' from JSON`);
}
