// @vercel/sdk/esm/models/putfirewallconfigop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const PutFirewallConfigAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsGenAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsXssAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsSfAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction = {
    Deny: "deny",
    Log: "log",
};
/**
 * [Parameter](https://vercel.com/docs/security/vercel-waf/rule-configuration#parameters) from the incoming traffic.
 */
export const PutFirewallConfigType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Region: "region",
    Protocol: "protocol",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
    BotName: "bot_name",
    BotCategory: "bot_category",
};
export const Op = {
    Re: "re",
    Eq: "eq",
    Neq: "neq",
    Ex: "ex",
    Nex: "nex",
    Inc: "inc",
    Ninc: "ninc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
};
export const PutFirewallConfigSecurityRequestRequestBodyRulesActionAction = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const Algo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const Action1 = {
    Log: "log",
    Challenge: "challenge",
    Deny: "deny",
    RateLimit: "rate_limit",
};
export const PutFirewallConfigSecurityRequestRequestBodyIpsAction = {
    Deny: "deny",
    Challenge: "challenge",
    Log: "log",
    Bypass: "bypass",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200Action = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction = {
    Deny: "deny",
    Log: "log",
};
export const PutFirewallConfigRulesType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Protocol: "protocol",
    Region: "region",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
    ServerAction: "server_action",
    BotName: "bot_name",
    BotCategory: "bot_category",
};
export const PutFirewallConfigRulesOp = {
    Re: "re",
    Eq: "eq",
    Ex: "ex",
    Inc: "inc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
    Nex: "nex",
    Ninc: "ninc",
    Neq: "neq",
};
export const PutFirewallConfigRulesSecurityResponse200Action = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const PutFirewallConfigRulesAlgo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const PutFirewallConfigRulesSecurityResponse200ApplicationJSONResponseBodyAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    RateLimit: "rate_limit",
};
export const PutFirewallConfigRulesSecurityType = {
    Host: "host",
    Path: "path",
    Method: "method",
    Header: "header",
    Query: "query",
    Cookie: "cookie",
    TargetPath: "target_path",
    Route: "route",
    RawPath: "raw_path",
    IpAddress: "ip_address",
    Protocol: "protocol",
    Region: "region",
    Scheme: "scheme",
    Environment: "environment",
    UserAgent: "user_agent",
    GeoContinent: "geo_continent",
    GeoCountry: "geo_country",
    GeoCountryRegion: "geo_country_region",
    GeoCity: "geo_city",
    GeoAsNumber: "geo_as_number",
    Ja4Digest: "ja4_digest",
    Ja3Digest: "ja3_digest",
    RateLimitApiId: "rate_limit_api_id",
    ServerAction: "server_action",
    BotName: "bot_name",
    BotCategory: "bot_category",
};
export const PutFirewallConfigRulesSecurityOp = {
    Re: "re",
    Eq: "eq",
    Ex: "ex",
    Inc: "inc",
    Pre: "pre",
    Suf: "suf",
    Sub: "sub",
    Gt: "gt",
    Gte: "gte",
    Lt: "lt",
    Lte: "lte",
    Nex: "nex",
    Ninc: "ninc",
    Neq: "neq",
};
export const PutFirewallConfigRulesSecurityResponseAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    Bypass: "bypass",
    RateLimit: "rate_limit",
    Redirect: "redirect",
};
export const PutFirewallConfigRulesSecurityAlgo = {
    FixedWindow: "fixed_window",
    TokenBucket: "token_bucket",
};
export const PutFirewallConfigRulesSecurityResponse200ApplicationJSONAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    RateLimit: "rate_limit",
};
export const PutFirewallConfigSecurityResponseAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
    Bypass: "bypass",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction = {
    Deny: "deny",
    Log: "log",
    Challenge: "challenge",
};
/** @internal */
export const ManagedRules$inboundSchema = z.object({});
/** @internal */
export const ManagedRules$outboundSchema = z.object({});
export function managedRulesToJSON(managedRules) {
    return JSON.stringify(ManagedRules$outboundSchema.parse(managedRules));
}
export function managedRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ManagedRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ManagedRules' from JSON`);
}
/** @internal */
export const PutFirewallConfigAction$inboundSchema = z.nativeEnum(PutFirewallConfigAction);
/** @internal */
export const PutFirewallConfigAction$outboundSchema = PutFirewallConfigAction$inboundSchema;
/** @internal */
export const Sd$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigAction$inboundSchema,
});
/** @internal */
export const Sd$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigAction$outboundSchema,
});
export function sdToJSON(sd) {
    return JSON.stringify(Sd$outboundSchema.parse(sd));
}
export function sdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Sd$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Sd' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityAction);
/** @internal */
export const PutFirewallConfigSecurityAction$outboundSchema = PutFirewallConfigSecurityAction$inboundSchema;
/** @internal */
export const Ma$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityAction$inboundSchema,
});
/** @internal */
export const Ma$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityAction$outboundSchema,
});
export function maToJSON(ma) {
    return JSON.stringify(Ma$outboundSchema.parse(ma));
}
export function maFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Ma$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Ma' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestAction);
/** @internal */
export const PutFirewallConfigSecurityRequestAction$outboundSchema = PutFirewallConfigSecurityRequestAction$inboundSchema;
/** @internal */
export const Lfi$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestAction$inboundSchema,
});
/** @internal */
export const Lfi$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestAction$outboundSchema,
});
export function lfiToJSON(lfi) {
    return JSON.stringify(Lfi$outboundSchema.parse(lfi));
}
export function lfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Lfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Lfi' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyAction$inboundSchema = z
    .nativeEnum(PutFirewallConfigSecurityRequestRequestBodyAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyAction$inboundSchema;
/** @internal */
export const Rfi$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyAction$inboundSchema,
});
/** @internal */
export const Rfi$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyAction$outboundSchema,
});
export function rfiToJSON(rfi) {
    return JSON.stringify(Rfi$outboundSchema.parse(rfi));
}
export function rfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Rfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Rfi' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsAction$inboundSchema;
/** @internal */
export const Rce$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsAction$inboundSchema,
});
/** @internal */
export const Rce$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsAction$outboundSchema,
});
export function rceToJSON(rce) {
    return JSON.stringify(Rce$outboundSchema.parse(rce));
}
export function rceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Rce$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Rce' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$inboundSchema;
/** @internal */
export const Php$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$inboundSchema,
});
/** @internal */
export const Php$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsPhpAction$outboundSchema,
});
export function phpToJSON(php) {
    return JSON.stringify(Php$outboundSchema.parse(php));
}
export function phpFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Php$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Php' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsGenAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$inboundSchema;
/** @internal */
export const Gen$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$inboundSchema,
});
/** @internal */
export const Gen$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsGenAction$outboundSchema,
});
export function genToJSON(gen) {
    return JSON.stringify(Gen$outboundSchema.parse(gen));
}
export function genFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Gen$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Gen' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsXssAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$inboundSchema;
/** @internal */
export const Xss$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$inboundSchema,
});
/** @internal */
export const Xss$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsXssAction$outboundSchema,
});
export function xssToJSON(xss) {
    return JSON.stringify(Xss$outboundSchema.parse(xss));
}
export function xssFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Xss$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Xss' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$inboundSchema;
/** @internal */
export const Sqli$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$inboundSchema,
});
/** @internal */
export const Sqli$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsSqliAction$outboundSchema,
});
export function sqliToJSON(sqli) {
    return JSON.stringify(Sqli$outboundSchema.parse(sqli));
}
export function sqliFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Sqli$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Sqli' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsSfAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$inboundSchema;
/** @internal */
export const Sf$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$inboundSchema,
});
/** @internal */
export const Sf$outboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsSfAction$outboundSchema,
});
export function sfToJSON(sf) {
    return JSON.stringify(Sf$outboundSchema.parse(sf));
}
export function sfFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Sf$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Sf' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$inboundSchema;
/** @internal */
export const Java$inboundSchema = z
    .object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$inboundSchema,
});
/** @internal */
export const Java$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityRequestRequestBodyCrsJavaAction$outboundSchema,
});
export function javaToJSON(java) {
    return JSON.stringify(Java$outboundSchema.parse(java));
}
export function javaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Java$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Java' from JSON`);
}
/** @internal */
export const Crs$inboundSchema = z
    .object({
    sd: z.lazy(() => Sd$inboundSchema).optional(),
    ma: z.lazy(() => Ma$inboundSchema).optional(),
    lfi: z.lazy(() => Lfi$inboundSchema).optional(),
    rfi: z.lazy(() => Rfi$inboundSchema).optional(),
    rce: z.lazy(() => Rce$inboundSchema).optional(),
    php: z.lazy(() => Php$inboundSchema).optional(),
    gen: z.lazy(() => Gen$inboundSchema).optional(),
    xss: z.lazy(() => Xss$inboundSchema).optional(),
    sqli: z.lazy(() => Sqli$inboundSchema).optional(),
    sf: z.lazy(() => Sf$inboundSchema).optional(),
    java: z.lazy(() => Java$inboundSchema).optional(),
});
/** @internal */
export const Crs$outboundSchema = z
    .object({
    sd: z.lazy(() => Sd$outboundSchema).optional(),
    ma: z.lazy(() => Ma$outboundSchema).optional(),
    lfi: z.lazy(() => Lfi$outboundSchema).optional(),
    rfi: z.lazy(() => Rfi$outboundSchema).optional(),
    rce: z.lazy(() => Rce$outboundSchema).optional(),
    php: z.lazy(() => Php$outboundSchema).optional(),
    gen: z.lazy(() => Gen$outboundSchema).optional(),
    xss: z.lazy(() => Xss$outboundSchema).optional(),
    sqli: z.lazy(() => Sqli$outboundSchema).optional(),
    sf: z.lazy(() => Sf$outboundSchema).optional(),
    java: z.lazy(() => Java$outboundSchema).optional(),
});
export function crsToJSON(crs) {
    return JSON.stringify(Crs$outboundSchema.parse(crs));
}
export function crsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Crs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Crs' from JSON`);
}
/** @internal */
export const PutFirewallConfigType$inboundSchema = z.nativeEnum(PutFirewallConfigType);
/** @internal */
export const PutFirewallConfigType$outboundSchema = PutFirewallConfigType$inboundSchema;
/** @internal */
export const Op$inboundSchema = z.nativeEnum(Op);
/** @internal */
export const Op$outboundSchema = Op$inboundSchema;
/** @internal */
export const Value$inboundSchema = z
    .union([z.string(), z.array(z.string()), z.number()]);
/** @internal */
export const Value$outboundSchema = z.union([z.string(), z.array(z.string()), z.number()]);
export function valueToJSON(value) {
    return JSON.stringify(Value$outboundSchema.parse(value));
}
export function valueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Value$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Value' from JSON`);
}
/** @internal */
export const Conditions$inboundSchema = z.object({
    type: PutFirewallConfigType$inboundSchema,
    op: Op$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.array(z.string()), z.number()]).optional(),
});
/** @internal */
export const Conditions$outboundSchema = z.object({
    type: PutFirewallConfigType$outboundSchema,
    op: Op$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.array(z.string()), z.number()]).optional(),
});
export function conditionsToJSON(conditions) {
    return JSON.stringify(Conditions$outboundSchema.parse(conditions));
}
export function conditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Conditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Conditions' from JSON`);
}
/** @internal */
export const ConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => Conditions$inboundSchema)),
});
/** @internal */
export const ConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => Conditions$outboundSchema)),
});
export function conditionGroupToJSON(conditionGroup) {
    return JSON.stringify(ConditionGroup$outboundSchema.parse(conditionGroup));
}
export function conditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ConditionGroup' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyRulesActionAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$inboundSchema;
/** @internal */
export const Algo$inboundSchema = z.nativeEnum(Algo);
/** @internal */
export const Algo$outboundSchema = Algo$inboundSchema;
/** @internal */
export const Action1$inboundSchema = z
    .nativeEnum(Action1);
/** @internal */
export const Action1$outboundSchema = Action1$inboundSchema;
/** @internal */
export const RateLimitAction$inboundSchema = z.union([Action1$inboundSchema, z.any()]);
/** @internal */
export const RateLimitAction$outboundSchema = z.union([Action1$outboundSchema, z.any()]);
export function rateLimitActionToJSON(rateLimitAction) {
    return JSON.stringify(RateLimitAction$outboundSchema.parse(rateLimitAction));
}
export function rateLimitActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RateLimitAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RateLimitAction' from JSON`);
}
/** @internal */
export const RateLimit1$inboundSchema = z.object({
    algo: Algo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(z.union([Action1$inboundSchema, z.any()])).optional(),
});
/** @internal */
export const RateLimit1$outboundSchema = z.object({
    algo: Algo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(z.union([Action1$outboundSchema, z.any()])).optional(),
});
export function rateLimit1ToJSON(rateLimit1) {
    return JSON.stringify(RateLimit1$outboundSchema.parse(rateLimit1));
}
export function rateLimit1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => RateLimit1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RateLimit1' from JSON`);
}
/** @internal */
export const RateLimit$inboundSchema = z.union([z.lazy(() => RateLimit1$inboundSchema), z.any()]);
/** @internal */
export const RateLimit$outboundSchema = z.union([z.lazy(() => RateLimit1$outboundSchema), z.any()]);
export function rateLimitToJSON(rateLimit) {
    return JSON.stringify(RateLimit$outboundSchema.parse(rateLimit));
}
export function rateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => RateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'RateLimit' from JSON`);
}
/** @internal */
export const Redirect1$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const Redirect1$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
export function redirect1ToJSON(redirect1) {
    return JSON.stringify(Redirect1$outboundSchema.parse(redirect1));
}
export function redirect1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => Redirect1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Redirect1' from JSON`);
}
/** @internal */
export const PutFirewallConfigRedirect$inboundSchema = z.union([z.lazy(() => Redirect1$inboundSchema), z.any()]);
/** @internal */
export const PutFirewallConfigRedirect$outboundSchema = z.union([z.lazy(() => Redirect1$outboundSchema), z.any()]);
export function putFirewallConfigRedirectToJSON(putFirewallConfigRedirect) {
    return JSON.stringify(PutFirewallConfigRedirect$outboundSchema.parse(putFirewallConfigRedirect));
}
export function putFirewallConfigRedirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRedirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRedirect' from JSON`);
}
/** @internal */
export const Mitigate$inboundSchema = z.object({
    action: PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$inboundSchema,
    rateLimit: z.nullable(z.union([z.lazy(() => RateLimit1$inboundSchema), z.any()])).optional(),
    redirect: z.nullable(z.union([z.lazy(() => Redirect1$inboundSchema), z.any()])).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const Mitigate$outboundSchema = z.object({
    action: PutFirewallConfigSecurityRequestRequestBodyRulesActionAction$outboundSchema,
    rateLimit: z.nullable(z.union([z.lazy(() => RateLimit1$outboundSchema), z.any()])).optional(),
    redirect: z.nullable(z.union([z.lazy(() => Redirect1$outboundSchema), z.any()])).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
export function mitigateToJSON(mitigate) {
    return JSON.stringify(Mitigate$outboundSchema.parse(mitigate));
}
export function mitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Mitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Mitigate' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyRulesAction$inboundSchema = z.object({
    mitigate: z.lazy(() => Mitigate$inboundSchema).optional(),
});
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyRulesAction$outboundSchema = z.object({
    mitigate: z.lazy(() => Mitigate$outboundSchema).optional(),
});
export function putFirewallConfigSecurityRequestRequestBodyRulesActionToJSON(putFirewallConfigSecurityRequestRequestBodyRulesAction) {
    return JSON.stringify(PutFirewallConfigSecurityRequestRequestBodyRulesAction$outboundSchema.parse(putFirewallConfigSecurityRequestRequestBodyRulesAction));
}
export function putFirewallConfigSecurityRequestRequestBodyRulesActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSecurityRequestRequestBodyRulesAction$inboundSchema
        .parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSecurityRequestRequestBodyRulesAction' from JSON`);
}
/** @internal */
export const ValidationErrors$inboundSchema = z.union([z.string(), z.array(z.string())]);
/** @internal */
export const ValidationErrors$outboundSchema = z.union([z.string(), z.array(z.string())]);
export function validationErrorsToJSON(validationErrors) {
    return JSON.stringify(ValidationErrors$outboundSchema.parse(validationErrors));
}
export function validationErrorsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => ValidationErrors$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ValidationErrors' from JSON`);
}
/** @internal */
export const PutFirewallConfigRules$inboundSchema = z.object({
    id: z.string().optional(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => ConditionGroup$inboundSchema)),
    action: z.lazy(() => PutFirewallConfigSecurityRequestRequestBodyRulesAction$inboundSchema),
    valid: z.boolean().optional(),
    validationErrors: z.union([z.string(), z.array(z.string())]).optional(),
});
/** @internal */
export const PutFirewallConfigRules$outboundSchema = z.object({
    id: z.string().optional(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => ConditionGroup$outboundSchema)),
    action: z.lazy(() => PutFirewallConfigSecurityRequestRequestBodyRulesAction$outboundSchema),
    valid: z.boolean().optional(),
    validationErrors: z.union([z.string(), z.array(z.string())]).optional(),
});
export function putFirewallConfigRulesToJSON(putFirewallConfigRules) {
    return JSON.stringify(PutFirewallConfigRules$outboundSchema.parse(putFirewallConfigRules));
}
export function putFirewallConfigRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRules' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyIpsAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityRequestRequestBodyIpsAction);
/** @internal */
export const PutFirewallConfigSecurityRequestRequestBodyIpsAction$outboundSchema = PutFirewallConfigSecurityRequestRequestBodyIpsAction$inboundSchema;
/** @internal */
export const Ips$inboundSchema = z
    .object({
    id: z.string().optional(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: PutFirewallConfigSecurityRequestRequestBodyIpsAction$inboundSchema,
});
/** @internal */
export const Ips$outboundSchema = z
    .object({
    id: z.string().optional(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: PutFirewallConfigSecurityRequestRequestBodyIpsAction$outboundSchema,
});
export function ipsToJSON(ips) {
    return JSON.stringify(Ips$outboundSchema.parse(ips));
}
export function ipsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Ips$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Ips' from JSON`);
}
/** @internal */
export const PutFirewallConfigRequestBody$inboundSchema = z.object({
    firewallEnabled: z.boolean(),
    managedRules: z.lazy(() => ManagedRules$inboundSchema).optional(),
    crs: z.lazy(() => Crs$inboundSchema).optional(),
    rules: z.array(z.lazy(() => PutFirewallConfigRules$inboundSchema)).optional(),
    ips: z.array(z.lazy(() => Ips$inboundSchema)).optional(),
    botIdEnabled: z.boolean().optional(),
});
/** @internal */
export const PutFirewallConfigRequestBody$outboundSchema = z.object({
    firewallEnabled: z.boolean(),
    managedRules: z.lazy(() => ManagedRules$outboundSchema).optional(),
    crs: z.lazy(() => Crs$outboundSchema).optional(),
    rules: z.array(z.lazy(() => PutFirewallConfigRules$outboundSchema))
        .optional(),
    ips: z.array(z.lazy(() => Ips$outboundSchema)).optional(),
    botIdEnabled: z.boolean().optional(),
});
export function putFirewallConfigRequestBodyToJSON(putFirewallConfigRequestBody) {
    return JSON.stringify(PutFirewallConfigRequestBody$outboundSchema.parse(putFirewallConfigRequestBody));
}
export function putFirewallConfigRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRequestBody' from JSON`);
}
/** @internal */
export const PutFirewallConfigRequest$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => PutFirewallConfigRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const PutFirewallConfigRequest$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => PutFirewallConfigRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function putFirewallConfigRequestToJSON(putFirewallConfigRequest) {
    return JSON.stringify(PutFirewallConfigRequest$outboundSchema.parse(putFirewallConfigRequest));
}
export function putFirewallConfigRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRequest' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$inboundSchema;
/** @internal */
export const PutFirewallConfigSd$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigSd$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSdAction$outboundSchema,
});
export function putFirewallConfigSdToJSON(putFirewallConfigSd) {
    return JSON.stringify(PutFirewallConfigSd$outboundSchema.parse(putFirewallConfigSd));
}
export function putFirewallConfigSdFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSd$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSd' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200Action$inboundSchema = z
    .nativeEnum(PutFirewallConfigSecurityResponse200Action);
/** @internal */
export const PutFirewallConfigSecurityResponse200Action$outboundSchema = PutFirewallConfigSecurityResponse200Action$inboundSchema;
/** @internal */
export const PutFirewallConfigMa$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200Action$inboundSchema,
});
/** @internal */
export const PutFirewallConfigMa$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200Action$outboundSchema,
});
export function putFirewallConfigMaToJSON(putFirewallConfigMa) {
    return JSON.stringify(PutFirewallConfigMa$outboundSchema.parse(putFirewallConfigMa));
}
export function putFirewallConfigMaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigMa$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigMa' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema;
/** @internal */
export const PutFirewallConfigLfi$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigLfi$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONAction$outboundSchema,
});
export function putFirewallConfigLfiToJSON(putFirewallConfigLfi) {
    return JSON.stringify(PutFirewallConfigLfi$outboundSchema.parse(putFirewallConfigLfi));
}
export function putFirewallConfigLfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigLfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigLfi' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema;
/** @internal */
export const PutFirewallConfigRfi$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigRfi$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema,
});
export function putFirewallConfigRfiToJSON(putFirewallConfigRfi) {
    return JSON.stringify(PutFirewallConfigRfi$outboundSchema.parse(putFirewallConfigRfi));
}
export function putFirewallConfigRfiFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRfi$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRfi' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$inboundSchema;
/** @internal */
export const PutFirewallConfigRce$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigRce$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveAction$outboundSchema,
});
export function putFirewallConfigRceToJSON(putFirewallConfigRce) {
    return JSON.stringify(PutFirewallConfigRce$outboundSchema.parse(putFirewallConfigRce));
}
export function putFirewallConfigRceFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRce$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRce' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$inboundSchema;
/** @internal */
export const PutFirewallConfigPhp$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigPhp$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsAction$outboundSchema,
});
export function putFirewallConfigPhpToJSON(putFirewallConfigPhp) {
    return JSON.stringify(PutFirewallConfigPhp$outboundSchema.parse(putFirewallConfigPhp));
}
export function putFirewallConfigPhpFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigPhp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigPhp' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$inboundSchema;
/** @internal */
export const PutFirewallConfigGen$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigGen$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsGenAction$outboundSchema,
});
export function putFirewallConfigGenToJSON(putFirewallConfigGen) {
    return JSON.stringify(PutFirewallConfigGen$outboundSchema.parse(putFirewallConfigGen));
}
export function putFirewallConfigGenFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigGen$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigGen' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$inboundSchema;
/** @internal */
export const PutFirewallConfigXss$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigXss$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsXssAction$outboundSchema,
});
export function putFirewallConfigXssToJSON(putFirewallConfigXss) {
    return JSON.stringify(PutFirewallConfigXss$outboundSchema.parse(putFirewallConfigXss));
}
export function putFirewallConfigXssFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigXss$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigXss' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$inboundSchema;
/** @internal */
export const PutFirewallConfigSqli$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigSqli$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSqliAction$outboundSchema,
});
export function putFirewallConfigSqliToJSON(putFirewallConfigSqli) {
    return JSON.stringify(PutFirewallConfigSqli$outboundSchema.parse(putFirewallConfigSqli));
}
export function putFirewallConfigSqliFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSqli$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSqli' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$inboundSchema;
/** @internal */
export const PutFirewallConfigSf$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigSf$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsSfAction$outboundSchema,
});
export function putFirewallConfigSfToJSON(putFirewallConfigSf) {
    return JSON.stringify(PutFirewallConfigSf$outboundSchema.parse(putFirewallConfigSf));
}
export function putFirewallConfigSfFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSf$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSf' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$inboundSchema;
/** @internal */
export const PutFirewallConfigJava$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigJava$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveCrsJavaAction$outboundSchema,
});
export function putFirewallConfigJavaToJSON(putFirewallConfigJava) {
    return JSON.stringify(PutFirewallConfigJava$outboundSchema.parse(putFirewallConfigJava));
}
export function putFirewallConfigJavaFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigJava$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigJava' from JSON`);
}
/** @internal */
export const PutFirewallConfigCrs$inboundSchema = z.object({
    sd: z.lazy(() => PutFirewallConfigSd$inboundSchema),
    ma: z.lazy(() => PutFirewallConfigMa$inboundSchema),
    lfi: z.lazy(() => PutFirewallConfigLfi$inboundSchema),
    rfi: z.lazy(() => PutFirewallConfigRfi$inboundSchema),
    rce: z.lazy(() => PutFirewallConfigRce$inboundSchema),
    php: z.lazy(() => PutFirewallConfigPhp$inboundSchema),
    gen: z.lazy(() => PutFirewallConfigGen$inboundSchema),
    xss: z.lazy(() => PutFirewallConfigXss$inboundSchema),
    sqli: z.lazy(() => PutFirewallConfigSqli$inboundSchema),
    sf: z.lazy(() => PutFirewallConfigSf$inboundSchema),
    java: z.lazy(() => PutFirewallConfigJava$inboundSchema),
});
/** @internal */
export const PutFirewallConfigCrs$outboundSchema = z.object({
    sd: z.lazy(() => PutFirewallConfigSd$outboundSchema),
    ma: z.lazy(() => PutFirewallConfigMa$outboundSchema),
    lfi: z.lazy(() => PutFirewallConfigLfi$outboundSchema),
    rfi: z.lazy(() => PutFirewallConfigRfi$outboundSchema),
    rce: z.lazy(() => PutFirewallConfigRce$outboundSchema),
    php: z.lazy(() => PutFirewallConfigPhp$outboundSchema),
    gen: z.lazy(() => PutFirewallConfigGen$outboundSchema),
    xss: z.lazy(() => PutFirewallConfigXss$outboundSchema),
    sqli: z.lazy(() => PutFirewallConfigSqli$outboundSchema),
    sf: z.lazy(() => PutFirewallConfigSf$outboundSchema),
    java: z.lazy(() => PutFirewallConfigJava$outboundSchema),
});
export function putFirewallConfigCrsToJSON(putFirewallConfigCrs) {
    return JSON.stringify(PutFirewallConfigCrs$outboundSchema.parse(putFirewallConfigCrs));
}
export function putFirewallConfigCrsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigCrs$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigCrs' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesType$inboundSchema = z.nativeEnum(PutFirewallConfigRulesType);
/** @internal */
export const PutFirewallConfigRulesType$outboundSchema = PutFirewallConfigRulesType$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesOp$inboundSchema = z.nativeEnum(PutFirewallConfigRulesOp);
/** @internal */
export const PutFirewallConfigRulesOp$outboundSchema = PutFirewallConfigRulesOp$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesValue$inboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
/** @internal */
export const PutFirewallConfigRulesValue$outboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
export function putFirewallConfigRulesValueToJSON(putFirewallConfigRulesValue) {
    return JSON.stringify(PutFirewallConfigRulesValue$outboundSchema.parse(putFirewallConfigRulesValue));
}
export function putFirewallConfigRulesValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesValue' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesConditions$inboundSchema = z.object({
    type: PutFirewallConfigRulesType$inboundSchema,
    op: PutFirewallConfigRulesOp$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
/** @internal */
export const PutFirewallConfigRulesConditions$outboundSchema = z.object({
    type: PutFirewallConfigRulesType$outboundSchema,
    op: PutFirewallConfigRulesOp$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
export function putFirewallConfigRulesConditionsToJSON(putFirewallConfigRulesConditions) {
    return JSON.stringify(PutFirewallConfigRulesConditions$outboundSchema.parse(putFirewallConfigRulesConditions));
}
export function putFirewallConfigRulesConditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesConditions' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => PutFirewallConfigRulesConditions$inboundSchema)),
});
/** @internal */
export const PutFirewallConfigRulesConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => PutFirewallConfigRulesConditions$outboundSchema)),
});
export function putFirewallConfigRulesConditionGroupToJSON(putFirewallConfigRulesConditionGroup) {
    return JSON.stringify(PutFirewallConfigRulesConditionGroup$outboundSchema.parse(putFirewallConfigRulesConditionGroup));
}
export function putFirewallConfigRulesConditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesConditionGroup' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesSecurityResponse200Action$inboundSchema = z
    .nativeEnum(PutFirewallConfigRulesSecurityResponse200Action);
/** @internal */
export const PutFirewallConfigRulesSecurityResponse200Action$outboundSchema = PutFirewallConfigRulesSecurityResponse200Action$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesAlgo$inboundSchema = z.nativeEnum(PutFirewallConfigRulesAlgo);
/** @internal */
export const PutFirewallConfigRulesAlgo$outboundSchema = PutFirewallConfigRulesAlgo$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema = z.nativeEnum(PutFirewallConfigRulesSecurityResponse200ApplicationJSONResponseBodyAction);
/** @internal */
export const PutFirewallConfigRulesSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema = PutFirewallConfigRulesSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesRateLimit$inboundSchema = z.object({
    algo: PutFirewallConfigRulesAlgo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(PutFirewallConfigRulesSecurityResponse200ApplicationJSONResponseBodyAction$inboundSchema).optional(),
});
/** @internal */
export const PutFirewallConfigRulesRateLimit$outboundSchema = z.object({
    algo: PutFirewallConfigRulesAlgo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(PutFirewallConfigRulesSecurityResponse200ApplicationJSONResponseBodyAction$outboundSchema).optional(),
});
export function putFirewallConfigRulesRateLimitToJSON(putFirewallConfigRulesRateLimit) {
    return JSON.stringify(PutFirewallConfigRulesRateLimit$outboundSchema.parse(putFirewallConfigRulesRateLimit));
}
export function putFirewallConfigRulesRateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesRateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesRateLimit' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesRedirect$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const PutFirewallConfigRulesRedirect$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
export function putFirewallConfigRulesRedirectToJSON(putFirewallConfigRulesRedirect) {
    return JSON.stringify(PutFirewallConfigRulesRedirect$outboundSchema.parse(putFirewallConfigRulesRedirect));
}
export function putFirewallConfigRulesRedirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesRedirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesRedirect' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesMitigate$inboundSchema = z.object({
    action: PutFirewallConfigRulesSecurityResponse200Action$inboundSchema,
    rateLimit: z.nullable(z.lazy(() => PutFirewallConfigRulesRateLimit$inboundSchema)).optional(),
    redirect: z.nullable(z.lazy(() => PutFirewallConfigRulesRedirect$inboundSchema)).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const PutFirewallConfigRulesMitigate$outboundSchema = z.object({
    action: PutFirewallConfigRulesSecurityResponse200Action$outboundSchema,
    rateLimit: z.nullable(z.lazy(() => PutFirewallConfigRulesRateLimit$outboundSchema)).optional(),
    redirect: z.nullable(z.lazy(() => PutFirewallConfigRulesRedirect$outboundSchema)).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
export function putFirewallConfigRulesMitigateToJSON(putFirewallConfigRulesMitigate) {
    return JSON.stringify(PutFirewallConfigRulesMitigate$outboundSchema.parse(putFirewallConfigRulesMitigate));
}
export function putFirewallConfigRulesMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesMitigate' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesAction$inboundSchema = z.object({
    mitigate: z.lazy(() => PutFirewallConfigRulesMitigate$inboundSchema)
        .optional(),
});
/** @internal */
export const PutFirewallConfigRulesAction$outboundSchema = z.object({
    mitigate: z.lazy(() => PutFirewallConfigRulesMitigate$outboundSchema)
        .optional(),
});
export function putFirewallConfigRulesActionToJSON(putFirewallConfigRulesAction) {
    return JSON.stringify(PutFirewallConfigRulesAction$outboundSchema.parse(putFirewallConfigRulesAction));
}
export function putFirewallConfigRulesActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesAction' from JSON`);
}
/** @internal */
export const PutFirewallConfigRules2$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => PutFirewallConfigRulesConditionGroup$inboundSchema)),
    action: z.lazy(() => PutFirewallConfigRulesAction$inboundSchema),
    valid: z.boolean(),
    validationErrors: z.array(z.string()),
});
/** @internal */
export const PutFirewallConfigRules2$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => PutFirewallConfigRulesConditionGroup$outboundSchema)),
    action: z.lazy(() => PutFirewallConfigRulesAction$outboundSchema),
    valid: z.boolean(),
    validationErrors: z.array(z.string()),
});
export function putFirewallConfigRules2ToJSON(putFirewallConfigRules2) {
    return JSON.stringify(PutFirewallConfigRules2$outboundSchema.parse(putFirewallConfigRules2));
}
export function putFirewallConfigRules2FromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRules2$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRules2' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesSecurityType$inboundSchema = z.nativeEnum(PutFirewallConfigRulesSecurityType);
/** @internal */
export const PutFirewallConfigRulesSecurityType$outboundSchema = PutFirewallConfigRulesSecurityType$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesSecurityOp$inboundSchema = z.nativeEnum(PutFirewallConfigRulesSecurityOp);
/** @internal */
export const PutFirewallConfigRulesSecurityOp$outboundSchema = PutFirewallConfigRulesSecurityOp$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesSecurityValue$inboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
/** @internal */
export const PutFirewallConfigRulesSecurityValue$outboundSchema = z.union([z.string(), z.number(), z.array(z.string())]);
export function putFirewallConfigRulesSecurityValueToJSON(putFirewallConfigRulesSecurityValue) {
    return JSON.stringify(PutFirewallConfigRulesSecurityValue$outboundSchema.parse(putFirewallConfigRulesSecurityValue));
}
export function putFirewallConfigRulesSecurityValueFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesSecurityValue$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesSecurityValue' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesSecurityConditions$inboundSchema = z.object({
    type: PutFirewallConfigRulesSecurityType$inboundSchema,
    op: PutFirewallConfigRulesSecurityOp$inboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
/** @internal */
export const PutFirewallConfigRulesSecurityConditions$outboundSchema = z.object({
    type: PutFirewallConfigRulesSecurityType$outboundSchema,
    op: PutFirewallConfigRulesSecurityOp$outboundSchema,
    neg: z.boolean().optional(),
    key: z.string().optional(),
    value: z.union([z.string(), z.number(), z.array(z.string())]).optional(),
});
export function putFirewallConfigRulesSecurityConditionsToJSON(putFirewallConfigRulesSecurityConditions) {
    return JSON.stringify(PutFirewallConfigRulesSecurityConditions$outboundSchema.parse(putFirewallConfigRulesSecurityConditions));
}
export function putFirewallConfigRulesSecurityConditionsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesSecurityConditions$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesSecurityConditions' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesSecurityConditionGroup$inboundSchema = z.object({
    conditions: z.array(z.lazy(() => PutFirewallConfigRulesSecurityConditions$inboundSchema)),
});
/** @internal */
export const PutFirewallConfigRulesSecurityConditionGroup$outboundSchema = z.object({
    conditions: z.array(z.lazy(() => PutFirewallConfigRulesSecurityConditions$outboundSchema)),
});
export function putFirewallConfigRulesSecurityConditionGroupToJSON(putFirewallConfigRulesSecurityConditionGroup) {
    return JSON.stringify(PutFirewallConfigRulesSecurityConditionGroup$outboundSchema.parse(putFirewallConfigRulesSecurityConditionGroup));
}
export function putFirewallConfigRulesSecurityConditionGroupFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesSecurityConditionGroup$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesSecurityConditionGroup' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesSecurityResponseAction$inboundSchema = z
    .nativeEnum(PutFirewallConfigRulesSecurityResponseAction);
/** @internal */
export const PutFirewallConfigRulesSecurityResponseAction$outboundSchema = PutFirewallConfigRulesSecurityResponseAction$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesSecurityAlgo$inboundSchema = z.nativeEnum(PutFirewallConfigRulesSecurityAlgo);
/** @internal */
export const PutFirewallConfigRulesSecurityAlgo$outboundSchema = PutFirewallConfigRulesSecurityAlgo$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesSecurityResponse200ApplicationJSONAction$inboundSchema = z.nativeEnum(PutFirewallConfigRulesSecurityResponse200ApplicationJSONAction);
/** @internal */
export const PutFirewallConfigRulesSecurityResponse200ApplicationJSONAction$outboundSchema = PutFirewallConfigRulesSecurityResponse200ApplicationJSONAction$inboundSchema;
/** @internal */
export const PutFirewallConfigRulesSecurityRateLimit$inboundSchema = z.object({
    algo: PutFirewallConfigRulesSecurityAlgo$inboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(PutFirewallConfigRulesSecurityResponse200ApplicationJSONAction$inboundSchema).optional(),
});
/** @internal */
export const PutFirewallConfigRulesSecurityRateLimit$outboundSchema = z.object({
    algo: PutFirewallConfigRulesSecurityAlgo$outboundSchema,
    window: z.number(),
    limit: z.number(),
    keys: z.array(z.string()),
    action: z.nullable(PutFirewallConfigRulesSecurityResponse200ApplicationJSONAction$outboundSchema).optional(),
});
export function putFirewallConfigRulesSecurityRateLimitToJSON(putFirewallConfigRulesSecurityRateLimit) {
    return JSON.stringify(PutFirewallConfigRulesSecurityRateLimit$outboundSchema.parse(putFirewallConfigRulesSecurityRateLimit));
}
export function putFirewallConfigRulesSecurityRateLimitFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesSecurityRateLimit$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesSecurityRateLimit' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesSecurityRedirect$inboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
/** @internal */
export const PutFirewallConfigRulesSecurityRedirect$outboundSchema = z.object({
    location: z.string(),
    permanent: z.boolean(),
});
export function putFirewallConfigRulesSecurityRedirectToJSON(putFirewallConfigRulesSecurityRedirect) {
    return JSON.stringify(PutFirewallConfigRulesSecurityRedirect$outboundSchema.parse(putFirewallConfigRulesSecurityRedirect));
}
export function putFirewallConfigRulesSecurityRedirectFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesSecurityRedirect$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesSecurityRedirect' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesSecurityMitigate$inboundSchema = z.object({
    action: PutFirewallConfigRulesSecurityResponseAction$inboundSchema,
    rateLimit: z.nullable(z.lazy(() => PutFirewallConfigRulesSecurityRateLimit$inboundSchema)).optional(),
    redirect: z.nullable(z.lazy(() => PutFirewallConfigRulesSecurityRedirect$inboundSchema)).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
/** @internal */
export const PutFirewallConfigRulesSecurityMitigate$outboundSchema = z.object({
    action: PutFirewallConfigRulesSecurityResponseAction$outboundSchema,
    rateLimit: z.nullable(z.lazy(() => PutFirewallConfigRulesSecurityRateLimit$outboundSchema)).optional(),
    redirect: z.nullable(z.lazy(() => PutFirewallConfigRulesSecurityRedirect$outboundSchema)).optional(),
    actionDuration: z.nullable(z.string()).optional(),
    bypassSystem: z.nullable(z.boolean()).optional(),
});
export function putFirewallConfigRulesSecurityMitigateToJSON(putFirewallConfigRulesSecurityMitigate) {
    return JSON.stringify(PutFirewallConfigRulesSecurityMitigate$outboundSchema.parse(putFirewallConfigRulesSecurityMitigate));
}
export function putFirewallConfigRulesSecurityMitigateFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesSecurityMitigate$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesSecurityMitigate' from JSON`);
}
/** @internal */
export const PutFirewallConfigRulesSecurityAction$inboundSchema = z.object({
    mitigate: z.lazy(() => PutFirewallConfigRulesSecurityMitigate$inboundSchema)
        .optional(),
});
/** @internal */
export const PutFirewallConfigRulesSecurityAction$outboundSchema = z.object({
    mitigate: z.lazy(() => PutFirewallConfigRulesSecurityMitigate$outboundSchema)
        .optional(),
});
export function putFirewallConfigRulesSecurityActionToJSON(putFirewallConfigRulesSecurityAction) {
    return JSON.stringify(PutFirewallConfigRulesSecurityAction$outboundSchema.parse(putFirewallConfigRulesSecurityAction));
}
export function putFirewallConfigRulesSecurityActionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRulesSecurityAction$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRulesSecurityAction' from JSON`);
}
/** @internal */
export const PutFirewallConfigRules1$inboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => PutFirewallConfigRulesSecurityConditionGroup$inboundSchema)),
    action: z.lazy(() => PutFirewallConfigRulesSecurityAction$inboundSchema),
    valid: z.boolean(),
    validationErrors: z.nullable(z.any()).optional(),
});
/** @internal */
export const PutFirewallConfigRules1$outboundSchema = z.object({
    id: z.string(),
    name: z.string(),
    description: z.string().optional(),
    active: z.boolean(),
    conditionGroup: z.array(z.lazy(() => PutFirewallConfigRulesSecurityConditionGroup$outboundSchema)),
    action: z.lazy(() => PutFirewallConfigRulesSecurityAction$outboundSchema),
    valid: z.boolean(),
    validationErrors: z.nullable(z.any()).optional(),
});
export function putFirewallConfigRules1ToJSON(putFirewallConfigRules1) {
    return JSON.stringify(PutFirewallConfigRules1$outboundSchema.parse(putFirewallConfigRules1));
}
export function putFirewallConfigRules1FromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigRules1$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigRules1' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityRules$inboundSchema = z.union([
    z.lazy(() => PutFirewallConfigRules1$inboundSchema),
    z.lazy(() => PutFirewallConfigRules2$inboundSchema),
]);
/** @internal */
export const PutFirewallConfigSecurityRules$outboundSchema = z.union([
    z.lazy(() => PutFirewallConfigRules1$outboundSchema),
    z.lazy(() => PutFirewallConfigRules2$outboundSchema),
]);
export function putFirewallConfigSecurityRulesToJSON(putFirewallConfigSecurityRules) {
    return JSON.stringify(PutFirewallConfigSecurityRules$outboundSchema.parse(putFirewallConfigSecurityRules));
}
export function putFirewallConfigSecurityRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigSecurityRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigSecurityRules' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponseAction$inboundSchema = z
    .nativeEnum(PutFirewallConfigSecurityResponseAction);
/** @internal */
export const PutFirewallConfigSecurityResponseAction$outboundSchema = PutFirewallConfigSecurityResponseAction$inboundSchema;
/** @internal */
export const PutFirewallConfigIps$inboundSchema = z.object({
    id: z.string(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: PutFirewallConfigSecurityResponseAction$inboundSchema,
});
/** @internal */
export const PutFirewallConfigIps$outboundSchema = z.object({
    id: z.string(),
    hostname: z.string(),
    ip: z.string(),
    notes: z.string().optional(),
    action: PutFirewallConfigSecurityResponseAction$outboundSchema,
});
export function putFirewallConfigIpsToJSON(putFirewallConfigIps) {
    return JSON.stringify(PutFirewallConfigIps$outboundSchema.parse(putFirewallConfigIps));
}
export function putFirewallConfigIpsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigIps$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigIps' from JSON`);
}
/** @internal */
export const PutFirewallConfigChanges$inboundSchema = z.object({});
/** @internal */
export const PutFirewallConfigChanges$outboundSchema = z.object({});
export function putFirewallConfigChangesToJSON(putFirewallConfigChanges) {
    return JSON.stringify(PutFirewallConfigChanges$outboundSchema.parse(putFirewallConfigChanges));
}
export function putFirewallConfigChangesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigChanges$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigChanges' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$inboundSchema;
/** @internal */
export const PutFirewallConfigBotProtection$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const PutFirewallConfigBotProtection$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
export function putFirewallConfigBotProtectionToJSON(putFirewallConfigBotProtection) {
    return JSON.stringify(PutFirewallConfigBotProtection$outboundSchema.parse(putFirewallConfigBotProtection));
}
export function putFirewallConfigBotProtectionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigBotProtection$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigBotProtection' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$inboundSchema;
/** @internal */
export const PutFirewallConfigAiBots$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const PutFirewallConfigAiBots$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesAiBotsAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
export function putFirewallConfigAiBotsToJSON(putFirewallConfigAiBots) {
    return JSON.stringify(PutFirewallConfigAiBots$outboundSchema.parse(putFirewallConfigAiBots));
}
export function putFirewallConfigAiBotsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigAiBots$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigAiBots' from JSON`);
}
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$inboundSchema = z.nativeEnum(PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction);
/** @internal */
export const PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$outboundSchema = PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$inboundSchema;
/** @internal */
export const PutFirewallConfigOwasp$inboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$inboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
/** @internal */
export const PutFirewallConfigOwasp$outboundSchema = z.object({
    active: z.boolean(),
    action: PutFirewallConfigSecurityResponse200ApplicationJSONResponseBodyActiveManagedRulesOwaspAction$outboundSchema
        .optional(),
    updatedAt: z.string().optional(),
    userId: z.string().optional(),
    username: z.string().optional(),
});
export function putFirewallConfigOwaspToJSON(putFirewallConfigOwasp) {
    return JSON.stringify(PutFirewallConfigOwasp$outboundSchema.parse(putFirewallConfigOwasp));
}
export function putFirewallConfigOwaspFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigOwasp$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigOwasp' from JSON`);
}
/** @internal */
export const PutFirewallConfigManagedRules$inboundSchema = z.object({
    bot_protection: z.lazy(() => PutFirewallConfigBotProtection$inboundSchema)
        .optional(),
    ai_bots: z.lazy(() => PutFirewallConfigAiBots$inboundSchema).optional(),
    owasp: z.lazy(() => PutFirewallConfigOwasp$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "bot_protection": "botProtection",
        "ai_bots": "aiBots",
    });
});
/** @internal */
export const PutFirewallConfigManagedRules$outboundSchema = z.object({
    botProtection: z.lazy(() => PutFirewallConfigBotProtection$outboundSchema)
        .optional(),
    aiBots: z.lazy(() => PutFirewallConfigAiBots$outboundSchema).optional(),
    owasp: z.lazy(() => PutFirewallConfigOwasp$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        botProtection: "bot_protection",
        aiBots: "ai_bots",
    });
});
export function putFirewallConfigManagedRulesToJSON(putFirewallConfigManagedRules) {
    return JSON.stringify(PutFirewallConfigManagedRules$outboundSchema.parse(putFirewallConfigManagedRules));
}
export function putFirewallConfigManagedRulesFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigManagedRules$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigManagedRules' from JSON`);
}
/** @internal */
export const Active$inboundSchema = z
    .object({
    ownerId: z.string(),
    projectKey: z.string(),
    id: z.string(),
    version: z.number(),
    updatedAt: z.string(),
    firewallEnabled: z.boolean(),
    crs: z.lazy(() => PutFirewallConfigCrs$inboundSchema),
    rules: z.array(z.union([
        z.lazy(() => PutFirewallConfigRules1$inboundSchema),
        z.lazy(() => PutFirewallConfigRules2$inboundSchema),
    ])),
    ips: z.array(z.lazy(() => PutFirewallConfigIps$inboundSchema)),
    changes: z.array(z.lazy(() => PutFirewallConfigChanges$inboundSchema)),
    managedRules: z.lazy(() => PutFirewallConfigManagedRules$inboundSchema)
        .optional(),
    botIdEnabled: z.boolean().optional(),
});
/** @internal */
export const Active$outboundSchema = z.object({
    ownerId: z.string(),
    projectKey: z.string(),
    id: z.string(),
    version: z.number(),
    updatedAt: z.string(),
    firewallEnabled: z.boolean(),
    crs: z.lazy(() => PutFirewallConfigCrs$outboundSchema),
    rules: z.array(z.union([
        z.lazy(() => PutFirewallConfigRules1$outboundSchema),
        z.lazy(() => PutFirewallConfigRules2$outboundSchema),
    ])),
    ips: z.array(z.lazy(() => PutFirewallConfigIps$outboundSchema)),
    changes: z.array(z.lazy(() => PutFirewallConfigChanges$outboundSchema)),
    managedRules: z.lazy(() => PutFirewallConfigManagedRules$outboundSchema)
        .optional(),
    botIdEnabled: z.boolean().optional(),
});
export function activeToJSON(active) {
    return JSON.stringify(Active$outboundSchema.parse(active));
}
export function activeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Active$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Active' from JSON`);
}
/** @internal */
export const PutFirewallConfigResponseBody$inboundSchema = z.object({
    active: z.lazy(() => Active$inboundSchema),
});
/** @internal */
export const PutFirewallConfigResponseBody$outboundSchema = z.object({
    active: z.lazy(() => Active$outboundSchema),
});
export function putFirewallConfigResponseBodyToJSON(putFirewallConfigResponseBody) {
    return JSON.stringify(PutFirewallConfigResponseBody$outboundSchema.parse(putFirewallConfigResponseBody));
}
export function putFirewallConfigResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PutFirewallConfigResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PutFirewallConfigResponseBody' from JSON`);
}
