// @vercel/sdk/esm/models/stageredirectsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const StatusCode$inboundSchema = z.union([z.number(), z.string()]);
/** @internal */
export const StatusCode$outboundSchema = z.union([z.number(), z.string()]);
export function statusCodeToJSON(statusCode) {
    return JSON.stringify(StatusCode$outboundSchema.parse(statusCode));
}
export function statusCodeFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StatusCode$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StatusCode' from JSON`);
}
/** @internal */
export const Redirects$inboundSchema = z.object({
    source: z.string(),
    destination: z.string(),
    statusCode: z.union([z.number(), z.string()]).optional(),
    permanent: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    query: z.boolean().optional(),
});
/** @internal */
export const Redirects$outboundSchema = z.object({
    source: z.string(),
    destination: z.string(),
    statusCode: z.union([z.number(), z.string()]).optional(),
    permanent: z.boolean().optional(),
    caseSensitive: z.boolean().optional(),
    query: z.boolean().optional(),
});
export function redirectsToJSON(redirects) {
    return JSON.stringify(Redirects$outboundSchema.parse(redirects));
}
export function redirectsFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Redirects$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Redirects' from JSON`);
}
/** @internal */
export const StageRedirectsRequestBody$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string(),
    overwrite: z.boolean().optional(),
    name: z.string().optional(),
    redirects: z.array(z.lazy(() => Redirects$inboundSchema)).optional(),
});
/** @internal */
export const StageRedirectsRequestBody$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string(),
    overwrite: z.boolean().optional(),
    name: z.string().optional(),
    redirects: z.array(z.lazy(() => Redirects$outboundSchema)).optional(),
});
export function stageRedirectsRequestBodyToJSON(stageRedirectsRequestBody) {
    return JSON.stringify(StageRedirectsRequestBody$outboundSchema.parse(stageRedirectsRequestBody));
}
export function stageRedirectsRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StageRedirectsRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StageRedirectsRequestBody' from JSON`);
}
/** @internal */
export const StageRedirectsRequest$inboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => StageRedirectsRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const StageRedirectsRequest$outboundSchema = z.object({
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => StageRedirectsRequestBody$outboundSchema)
        .optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function stageRedirectsRequestToJSON(stageRedirectsRequest) {
    return JSON.stringify(StageRedirectsRequest$outboundSchema.parse(stageRedirectsRequest));
}
export function stageRedirectsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StageRedirectsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StageRedirectsRequest' from JSON`);
}
/** @internal */
export const StageRedirectsVersion$inboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
/** @internal */
export const StageRedirectsVersion$outboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
export function stageRedirectsVersionToJSON(stageRedirectsVersion) {
    return JSON.stringify(StageRedirectsVersion$outboundSchema.parse(stageRedirectsVersion));
}
export function stageRedirectsVersionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StageRedirectsVersion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StageRedirectsVersion' from JSON`);
}
/** @internal */
export const StageRedirectsResponseBody$inboundSchema = z.object({
    alias: z.nullable(z.string()),
    version: z.lazy(() => StageRedirectsVersion$inboundSchema),
});
/** @internal */
export const StageRedirectsResponseBody$outboundSchema = z.object({
    alias: z.nullable(z.string()),
    version: z.lazy(() => StageRedirectsVersion$outboundSchema),
});
export function stageRedirectsResponseBodyToJSON(stageRedirectsResponseBody) {
    return JSON.stringify(StageRedirectsResponseBody$outboundSchema.parse(stageRedirectsResponseBody));
}
export function stageRedirectsResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => StageRedirectsResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'StageRedirectsResponseBody' from JSON`);
}
