// @vercel/sdk/esm/models/updateversionop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const UpdateVersionAction = {
    Promote: "promote",
    Restore: "restore",
    Discard: "discard",
};
/** @internal */
export const UpdateVersionAction$inboundSchema = z.nativeEnum(UpdateVersionAction);
/** @internal */
export const UpdateVersionAction$outboundSchema = UpdateVersionAction$inboundSchema;
/** @internal */
export const UpdateVersionRequestBody$inboundSchema = z.object({
    id: z.string(),
    action: UpdateVersionAction$inboundSchema,
    name: z.string().optional(),
});
/** @internal */
export const UpdateVersionRequestBody$outboundSchema = z.object({
    id: z.string(),
    action: UpdateVersionAction$outboundSchema,
    name: z.string().optional(),
});
export function updateVersionRequestBodyToJSON(updateVersionRequestBody) {
    return JSON.stringify(UpdateVersionRequestBody$outboundSchema.parse(updateVersionRequestBody));
}
export function updateVersionRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateVersionRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateVersionRequestBody' from JSON`);
}
/** @internal */
export const UpdateVersionRequest$inboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    RequestBody: z.lazy(() => UpdateVersionRequestBody$inboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateVersionRequest$outboundSchema = z.object({
    projectId: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
    requestBody: z.lazy(() => UpdateVersionRequestBody$outboundSchema).optional(),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
export function updateVersionRequestToJSON(updateVersionRequest) {
    return JSON.stringify(UpdateVersionRequest$outboundSchema.parse(updateVersionRequest));
}
export function updateVersionRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateVersionRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateVersionRequest' from JSON`);
}
/** @internal */
export const UpdateVersionVersion$inboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
/** @internal */
export const UpdateVersionVersion$outboundSchema = z.object({
    id: z.string(),
    key: z.string(),
    lastModified: z.number(),
    createdBy: z.string(),
    name: z.string().optional(),
    isStaging: z.boolean().optional(),
    isLive: z.boolean().optional(),
    redirectCount: z.number().optional(),
    alias: z.string().optional(),
});
export function updateVersionVersionToJSON(updateVersionVersion) {
    return JSON.stringify(UpdateVersionVersion$outboundSchema.parse(updateVersionVersion));
}
export function updateVersionVersionFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateVersionVersion$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateVersionVersion' from JSON`);
}
/** @internal */
export const UpdateVersionResponseBody$inboundSchema = z.object({
    version: z.lazy(() => UpdateVersionVersion$inboundSchema),
});
/** @internal */
export const UpdateVersionResponseBody$outboundSchema = z.object({
    version: z.lazy(() => UpdateVersionVersion$outboundSchema),
});
export function updateVersionResponseBodyToJSON(updateVersionResponseBody) {
    return JSON.stringify(UpdateVersionResponseBody$outboundSchema.parse(updateVersionResponseBody));
}
export function updateVersionResponseBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateVersionResponseBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateVersionResponseBody' from JSON`);
}
